observatory-dags/observatory/dags/telescopes/oapen_irus_uk.py

```diff
@@ -0,0 +1,459 @@
+            raise AirflowException('Transfer blob unsuccessful')
+
+    def download_transform(self):
+        """ Download blob with access stats to a local file.
+
+        :return: None.
+        """
+        success = download_blob_from_cloud_storage(self.download_bucket, self.blob_name, self.transform_path)
+        if not success:
+            raise AirflowException('Download blob unsuccessful')
+
+
+class OapenIrusUkTelescope(SnapshotTelescope):
+    DAG_ID_PREFIX = 'oapen_irus_uk'
+    ORG_MAPPING = {'ucl_press': quote("UCL Press"),
+                   'anu_press': quote("ANU Press"),
+                   'wits_university_press': quote("Wits University Press")}
+    CLOUD_FUNCTION_MD5_HASH = '61793f6e4864285088917b81186f125e'
+
+    def __init__(self, organisation: Organisation, dag_id: Optional[str] = None,
+                 start_date: datetime = datetime(2020, 2, 1),
+                 schedule_interval: str = '@monthly', dataset_id: str = 'oapen',
+                 dataset_description: str = 'Oapen dataset', catchup: bool = True, airflow_vars: List = None,
+                 airflow_conns: List = None,
+                 max_active_runs=5):
+
+        """ The OAPEN irus uk telescope.
+
+        :param dag_id: the id of the DAG.
+        :param start_date: the start date of the DAG.
+        :param schedule_interval: the schedule interval of the DAG.
+        :param dataset_id: the BigQuery dataset id.
+        :param dataset_description: description for the BigQuery dataset.
+        :param catchup:  whether to catchup the DAG or not.
+        :param airflow_vars: list of airflow variable keys, for each variable it is checked if it exists in airflow.
+        """
+
+        if airflow_vars is None:
+            airflow_vars = [AirflowVars.DATA_PATH, AirflowVars.PROJECT_ID, AirflowVars.DATA_LOCATION,
+                            AirflowVars.DOWNLOAD_BUCKET, AirflowVars.TRANSFORM_BUCKET]
+
+        if airflow_conns is None:
+            airflow_conns = [AirflowConns.GEOIP_LICENSE_KEY, AirflowConns.OAPEN_IRUS_UK_API,
+                             AirflowConns.OAPEN_IRUS_UK_LOGIN]
+
+        if dag_id is None:
+            dag_id = make_dag_id(self.DAG_ID_PREFIX, organisation.name)
+
+        super().__init__(dag_id, start_date, schedule_interval, dataset_id, dataset_description=dataset_description,
+                         catchup=catchup, airflow_vars=airflow_vars, airflow_conns=airflow_conns,
+                         max_active_runs=max_active_runs)
+        self.organisation = organisation
+        self.publisher_name = get_publisher_name(make_org_id(organisation.name))
+        self.oapen_project_id = 'oapen-usage-data-gdpr-proof'
+        self.function_name = 'oapen_access_stats'
+        self.function_region = 'europe-west1'
+        self.function_source_url = 'https://github.com/The-Academic-Observatory/oapen-irus-uk-cloud-function/releases' \
+                                   '/latest/download/oapen-irus-uk-cloud-function.zip '
```

Comment:
It would be good to store the string values as constants in the class, e.g. like DAG_ID_PREFIX